### DISCUSSION TITLE
Strategies for encrypted data across native&web platforms

### DISCUSSION THREAD

**Discussion by justro** [simolus3/drift#2618] [Q&A]
Hello - I've looked at the docs for encryption, and understand existing strategies apply to native platforms. I'd like to figure out a way to encrypt our data that will also include web. I'm assuming that probably the way to do it is to encrypt what we store, was hoping folks could share strategies for how they do that - is there a particular level where tying in makes the most sense within drift, or the table classes? TIA!

**Comment by simolus3** [comment]
Implementing encryption probably makes most sense on the fairly low-level virtual file system implementation part of the `sqlite3` package. That way you don't need to apply encryption in drift (like in a type converter for sensitive columns).

Since every website is sandbox and the database will have to be unencrypted in parts to live in memory, I wonder what the purpose of encrypting on the web is. Making reverse-engineering harder? I don't really see a security benefit here.

**Reply by justro** [reply]
My thinking was that, if any personal data were stored in the DB, and the browser is on a shared device where someone might be able to access the device's file system, it'd probably be best to have it encrypted. I had also considered maybe just using an in-memory db and making web users just connect to the servers more often, but if someone were running as a PWA on a device, in a perfect world, offline mode would work. I'd generally like to encrypt any user data at rest, regardless of it being in a browser or on my servers, etc. That said, if it's very difficult to make work, I'll probably come up with a different approach. 

I did look a bit at extensions for SQLite that may compile for web and might work, but I'm not sure if I can invest the effort right now. If I do end up digging into it, I'll share what I come up with for anyone interested. Thanks!
